### ISSUE TITLE
Keys / Secrets need to be stored in KeyVault

### ISSUE THREAD

**Issue by cleemullins** [microsoft/botframework-components#958]
## Describe the bug
Keys, related to my publishing profile, are stored in appsettings.json. These keys are unencrypted. This is unexpected, as I expected them to be stored in Key Vault. 

[Note: I had a bunch of screen shots around this, but they're not useful so I deleted them]

There is a .gitignore file, that prevents this file from being checked in:
![image](https://user-images.githubusercontent.com/1165321/115451460-1310bc00-a1d2-11eb-9b33-9527a720c3a5.png)

## Version

Version:  1.4.0-nightly.237101.f01cb52
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

## Expected behavior

The Azure Key Management Doc is clear:
https://docs.microsoft.com/en-us/azure/architecture/framework/security/design-storage-keys

> Store all application keys and secrets in managed key vault service such as Azure Key Vault. Data encryption keys are often encrypted with a key encryption key in Azure Key Vault to further limit access.
> 
> Make sure that no keys and secrets for any environment types (Dev/Test, or production) are stored in application configuration files or CI/CD pipelines. Developers can use Visual Studio Connected Services or local-only files to access credentials.

![image](https://user-images.githubusercontent.com/1165321/115451731-6aaf2780-a1d2-11eb-8d6e-540bb3c80345.png)

**Comment by tracyboehrer** [comment]
@cwhitten Why would this be in Components?  This is driven by Composer.  Unless there is also some discussion about moving provisioning and publishing out of Composer.

**Comment by cwhitten** [comment]
@darrenj & @garypretty can comment, but it was offered to decorate sensitive fields in appsettings with the keyvault alias, and we told not to because of changes needed in the runtime to properly consume the fields. The old runtime template had this working, by the way.

**Comment by tracyboehrer** [comment]
@darrenj & @garypretty What am I missing?  Runtime does even read (or write) those profile values in appsettings.  Honestly, appsettings isn't the right place for the publishing profile to begin with.  It has nothing to do with the running bot or runtime.

**Comment by cwhitten** [comment]
Let me clarify - the MSAppId and AppPassword _are_ fields in appsettings that the runtime reads from. You're right in that the publishing profile is not needed by the runtime.

**Comment by tracyboehrer** [comment]
Now I'm with you.

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by peterinnesmsft** [comment]
@cwhitten @tracyboehrer I think there is some confusion around Key Vault support in the runtime, and it sounds like that got miscommunicated to you Chris.

Chris, you are absolutely right - you should be able to store secrets in appsettings.json using the Key Vault references as described here: https://docs.microsoft.com/en-us/azure/app-service/app-service-key-vault-references. This is a capability that is offered up by App Service, and is not tied to the runtime implementation.

To my understanding, I believe that all that is required here is to ensure that there is an AKV resource that is granted access to your compute hosting your bot through a valid MSI. If that's in place, then App Service I believe under-the-hood does variable substitution, so as long as it can suitably detect the application settings and the resources are granted access, it should be able to fill them in.

---

However, the catch with this is that it is entirely dependent on being deployed to App Service (and by extension, Functions). If you chose to use a different compute offering, such as Service Fabric, AKS, etc., then you would not be able to use this particular syntax. This is where the the Azure Key Vault configuration provider comes in, as documented here: https://docs.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-3.1 This *would* be a runtime change, and is most likely what was being referred to.

I don't know that there has been any particular validation of App Service support for AKV configuration references, so I think this is something we should try and verify separately to adding AKV as an explicit configuration source.

**Comment by cwhitten** [comment]
Makes sense @peterinnesmsft. Our thinking has been to revisit this post BUILD, and make sure KV is plumbed through the stack in a sane way for customers.

I also realize that I transferred the wrong key vault github issue here, and may have avoided the confusion altogether had I moved the correct one.
